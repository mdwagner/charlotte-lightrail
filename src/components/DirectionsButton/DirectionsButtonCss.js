import styled from 'styled-components/native';

export const BubbleTouchableOpacity = styled.TouchableOpacity`
  backgroundColor: ${props => props.theme.backgroundColorButton};
  borderRadius: 50;
  borderWidth: 0;
  marginHorizontal: 6;
  padding: 7px;
  width: 50;
  height: 50;
`;

export const ActiveIconImage = styled.Image`
  opacity: 1;
`;
