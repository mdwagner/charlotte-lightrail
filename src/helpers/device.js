import { Dimensions } from 'react-native';

// identify iPhone model based on device width
const identifyDevice = (deviceScreenProps) => {
  if (deviceScreenProps.width === 414) {
    return 'iPhone 6+';
  } else if (deviceScreenProps.width === 320) {
    return 'iPhone 5';
  }
  return 'iPhone 6';
};

const deviceScreen = Dimensions.get('window');
const deviceName = identifyDevice(deviceScreen);
let deviceVariableSizes = {};
let defaultCenter = {};
let defaultZoom;
if (deviceName === 'iPhone 6+') {
  deviceVariableSizes = {
    calloutHeight: 300,
    nextTimeFontSize: 40,
    stationLabelFontSize: 36,
    nextTimeWrapPaddingVertical: 4,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.12848262558094,
    longitude: -80.84703007335676
  };
  defaultZoom = 10.35016331854935;
} else if (deviceName === 'iPhone 6') {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 36,
    stationLabelFontSize: 30,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest Station - '
  };
  defaultCenter = {
    latitude: 35.09018630471958,
    longitude: -80.84307324707783
  };
  defaultZoom = 9.228717656576594;
} else {
  deviceVariableSizes = {
    calloutHeight: 279,
    nextTimeFontSize: 33,
    stationLabelFontSize: 22,
    nextTimeWrapPaddingVertical: 1,
    nearestStationText: 'Nearest - '
  };
  defaultCenter = {
    latitude: 35.12642689061146,
    longitude: -80.8549019571087
  };
  defaultZoom = 10.10400032344327;
}

const calloutBoxHeight = deviceVariableSizes.calloutHeight - 25; // excludes triangle
const blueBoxHeight = calloutBoxHeight * 0.45; // 45% of calloutBoxHeight
const blackBoxHeight = calloutBoxHeight * 0.55; // 55% of calloutBoxHeight

const insideColPadding = deviceScreen.width * 0.13;
const circleWrapperHeight = 54;
const circleWidth = 40;

export const deviceProps = {
  calloutBoxHeight,
  blueBoxHeight,
  blackBoxHeight,
  deviceVariableSizes,
  deviceScreen,
  defaultCenter,
  defaultZoom,
  deviceName,
  insideColPadding,
  circleWrapperHeight,
  circleWidth
};
