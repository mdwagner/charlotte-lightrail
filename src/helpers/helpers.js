import { Linking } from 'react-native';
import { mapboxApiKey } from './config';

export const startNavigation = async (mode, latlng) => {
  const { latitude, longitude } = latlng;

  const googleMapsURL = `comgooglemaps://?daddr=${latitude},${longitude}&directionsmode=${mode}`;
  const appleMapsURL = `http://maps.apple.com?daddr=${latitude},${longitude}&dirflg=${mode}`;

  const supported = await Linking.canOpenURL(googleMapsURL);
  if (supported) {
    Linking
      .openURL(googleMapsURL)
      .catch(err => console.error('Either Google Maps is not installed or some other error occurred: ', err));
  } else {
    Linking
      .openURL(appleMapsURL)
      .catch(err => console.error('An error occurred: ', err));
  }
};

export const mapboxDistanceAPI = {
  async getDistance(origin, destinations, mode) {
    const coordinates = destinations;
    coordinates.unshift(origin);

    if (__DEV__) {
      const endpoint = `directions-matrix/v1/mapbox/${mode}`;
      coordinates_string = coordinates.map(coordinate => {
        return coordinate.toString();
      });
      coordinates_query = coordinates_string.join(';');
      const url = `https://api.mapbox.com/${endpoint}/${coordinates_query}?access_token=${mapboxApiKey}`;
      const res = await fetch(url);
      return res.json();
    } else {
      const endpoint = `distances/v1/mapbox/${mode}`;
      const url = `https://api.mapbox.com/${endpoint}?access_token=${mapboxApiKey}`;
      const res = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          coordinates,
        }),
      });
      return res.json();
    }
  },
};