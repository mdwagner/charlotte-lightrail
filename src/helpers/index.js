export { withHelpers, withStaticData } from './hoc';