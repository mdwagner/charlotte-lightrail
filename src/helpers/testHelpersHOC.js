import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import renderer from 'react-test-renderer';
import { COLORS } from '../assets/styles/constants';

export const snapshotStyledComponent = (component) => {
  const snapshot = renderer.create(
    <ThemeProvider theme={COLORS}>
      {component}
    </ThemeProvider>
  ).toJSON();
  expect(snapshot).toMatchSnapshot();
};

export const themeStyledComponent = component => (
  <ThemeProvider theme={COLORS}>
    {component}
  </ThemeProvider>
);
