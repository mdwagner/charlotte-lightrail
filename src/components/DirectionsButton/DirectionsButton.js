import React from 'react';
import PropTypes from 'prop-types';
import { BubbleTouchableOpacity, ActiveIconImage } from './DirectionsButtonCss';

const iconImage = require('../../assets/icons/directions/ic_directions_white_36pt.png');

export default class DirectionsButton extends React.Component {

  static propTypes = {
    onPress: PropTypes.func.isRequired
  }

  render() {
    const { onPress } = this.props;

    return (
      <BubbleTouchableOpacity onPress={onPress}>
        <ActiveIconImage source={iconImage} />
      </BubbleTouchableOpacity>
    );
  }
}
