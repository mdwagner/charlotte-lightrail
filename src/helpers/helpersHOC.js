import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';

export function withHelpers(Component) {
  return class extends React.Component {
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}
