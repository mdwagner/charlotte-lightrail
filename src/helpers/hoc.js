import React from 'react';
import * as configHelpers from './config';
import * as deviceHelpers from './device';
import * as helperHelpers from './helpers';
import * as mapSetupHelpers from './mapSetup';
import * as scheduleHelpers from './scheduleCalcs';
import staticData from './staticData.json';

export function withHelpers(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        ...configHelpers,
        ...deviceHelpers,
        ...helperHelpers,
        ...mapSetupHelpers,
        ...scheduleHelpers
      };
      return (
        <Component {...this.props} helpers={newProps} />
      );
    }
  };
}

export function withStaticData(Component) {
  return class extends React.Component {
    constructor(props) {
      super(props);
    }
    render() {
      const newProps = {
        staticData
      };
      return (
        <Component {...this.props} staticData={newProps} />
      );
    }
  };
}