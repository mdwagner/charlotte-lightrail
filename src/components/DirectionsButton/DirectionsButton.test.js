import React from 'react';
import { snapshotStyledComponent } from '../../helpers/testHelpersHOC';
import DirectionsButton from './DirectionsButton';

describe('<DirectionsButton />', () => {
  let props;

  beforeAll(() => {
    props = {
      onPress: jest.fn()
    };
  });

  it('renders correctly', () => {
    snapshotStyledComponent(
      <DirectionsButton {...props} />
    );
  });
});
