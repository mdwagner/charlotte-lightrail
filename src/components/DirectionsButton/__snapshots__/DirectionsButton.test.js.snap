exports[`<DirectionsButton /> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#009ada",
      "borderRadius": 50,
      "borderWidth": 0,
      "height": 50,
      "marginHorizontal": 6,
      "opacity": 1,
      "paddingBottom": 7,
      "paddingLeft": 7,
      "paddingRight": 7,
      "paddingTop": 7,
      "width": 50,
    }
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Array [
        Object {
          "opacity": 1,
        },
        undefined,
      ]
    } />
</View>
`;
